tiling.js

```diff
@@ -1476,19 +1483,46 @@ class Spaces extends Map {
         }
     }
 
-    selectSpace(direction, move, transition) {
+    _getSpacesStack(use_mru) {
+        let spaces = [];
+
+        // get current workspace and current workspaceindex
+        let currentSpace = this.spaceOf(workspaceManager.get_active_workspace());
+        let currentSpace_index = currentSpace.workspace.index();
+
+        // use the mru stack to get array of spaces
+        if (use_mru === true) {
+            spaces = [...this.stack];
+            // go over each monitor and collect spaces
+            this.monitors.forEach(space => spaces.splice(spaces.indexOf(space), 1));
```

Comment:
Right, this line isn't very obvious :laughing: It goes through the spaces that are visible on a monitor and removes them from the space array as we don't want to duplicate them.